src/pql/schema.py

```diff
@@ -0,0 +1,152 @@
+step_extract_http = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "extract"},
+        "method": {"enum": ["http.get", "http.post"]},
+        "uri": {"type": "string"},
+    },
+    "required": ["step", "method", "uri"],
+    "if": {"properties": {"method": {"const": "http.post"}}},
+    "then": {"properties": {"params": {"type": "object"}}, "required": ["params"]},
+}
+
+step_extract_sql = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "extract"},
+        "method": {"const": "sql.postgres"},
+        "uri": {"type": "string"},
+        "query": {"type": "string"},
+    },
+    "required": ["step", "method", "uri", "query"],
+}
+
+step_extract_eth = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "extract"},
+        "method": {"enum": ["eth.balance", "eth.function"]},
+        "address": {"type": "string"},
+    },
+    "required": ["step", "method", "address"],
+    "if": {"properties": {"method": {"const": "eth.balance"}}},
+    "then": {
+        "properties": {
+            "params": {
+                "type": "object",
+                "properties": {
+                    "block": {"oneOf": [{"const": "latest"}, {"type": "integer"}]},
+                    "num_confirmations": {"type": "integer"},
+                },
+                "required": ["block"],
+            }
+        }
+    },
+    "else": {
+        "properties": {
+            "params": {
+                "type": "object",
+                "properties": {
+                    "function": {"type": "string"},
+                    "args": {"type": "array", "items": {"type": "string"}},
+                    "num_confirmations": {"type": "integer"},
+                    "block": {"oneOf": [{"const": "latest"}, {"type": "integer"}]},
+                },
+                "required": ["function", "args", "block"],
+            },
+        },
+        "required": ["params"],
+    },
+}
+
+step_traverse = {
+    "type": "object",
+    "properties": {
+        "method": {"const": "json"},
+        "params": {"type": "array", "items": {"type": "string"}},
+    },
+    "required": ["method", "params"],
+}
+
+step_get_index = {
+    "type": "object",
+    "properties": {"step": {"const": "get_index"}, "params": {"type": "integer"}},
+    "required": ["step", "params"],
+}
+
+step_math = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "math"},
+        "method": {"enum": ["mul", "add", "sub", "div"]},
+        "params": {"type": "number"},
+        "direction": {"const": {"reverse"}},
+    },
+    "required": ["step", "method", "params"],
+}
+
+step_custom = {
```

Comment:
I believe this has to be documented somehow, probably best if the custom step schema has to be supplied along with the definition in `src.config.PQL_CUSTOM_METHODS` and then the `source` object below also appends the schemas found in this object. Maybe even better if a custom function is defined as a `CustomFunction` object or something similar to make it more readable. What do you think?

Right now the `examples/custom_function.py` won't work due to missing schema.

Reply:
I've made an attempt at your proposed implementation.  Good idea 👍

Reply:
I've created an alternative implementation that allows the user to specify the custom methods / plugs as text and then we import using importlib.  You can find that implementation [here](https://github.com/frisitano/paralink-node/blob/sklearn/src/config.py#L63).

Reply:
I like the multi-file approach with it. It is up to you, if you fancy it more, just submit another PR.